jupyter_server/services/contents/filemanager.py

```diff
@@ -82,14 +82,19 @@ def _validate_post_save_hook(self, proposal):
             value = import_item(value)
         if not callable(value):
             raise TraitError("post_save_hook must be callable")
+        self._post_save_hooks.append(value)
         return value
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._post_save_hooks = []
```

Comment:
do we need to define a traitlets type for this variable?

Reply:
I wanted to keep the same API (which uses a trait for `pre_save_hook` and `post_save_hook`), and not introduce e.g. `register_pre_save_hook`. So I `_pre_save_hooks` and `_post_save_hooks` are not part of the user API and should not be traits.

Reply:
okay.. probably if you can share a simple example to show, how someone could set multiple function as `_pre_save_hooks` and `_post_save_hooks` would help.

Reply:
The API is still:
```python
file_contents_manager.pre_save_hook = my_hook_1
file_contents_manager.pre_save_hook = my_hook_2
```
but instead of having `my_hook_2` override `my_hook_1`, both are registered and called sequentially before saving a file. Same for `post_save_hook`.

Reply:
thanks @davidbrochart  for sharing this!